src/test/java/net/minecraftforge/debug/DataGeneratorTest.java

```diff
@@ -136,54 +153,55 @@ protected void registerRecipes(Consumer<IFinishedRecipe> consumer)
+                                    .setGroup("")
+                                    .addCriterion("has_dirt",
+                                            hasItem(Blocks.DIRT)) // DUMMY: Necessary, but not used when a custom advancement is provided through setAdvancement
+                                    ::build
                     )
-                )
-                .addAdvancement(
-                    Advancement.Builder.builder()
-                    .withParentId(new ResourceLocation("minecraft", "root"))
-                    .withDisplay(Blocks.DIAMOND_BLOCK,
-                        new StringTextComponent("Dirt2Diamonds"),
-                        new StringTextComponent("The BEST crafting recipe in the game!"),
-                        null, FrameType.TASK, false, false, false
+                    .setAdvancement(ID,
+                            ConditionalAdvancement.builder()
+                                    .addCondition(
+                                            and(
+                                                    not(modLoaded("minecraft")),
+                                                    itemExists("minecraft", "dirt"),
+                                                    FALSE()
+                                            )
+                                    )
+                                    .addAdvancement(
+                                            Advancement.Builder.builder()
+                                                    .withParentId(new ResourceLocation("minecraft", "root"))
+                                                    .withDisplay(Blocks.DIAMOND_BLOCK,
+                                                            new StringTextComponent("Dirt2Diamonds"),
+                                                            new StringTextComponent("The BEST crafting recipe in the game!"),
+                                                            null, FrameType.TASK, false, false, false
+                                                    )
+                                                    .withRewards(AdvancementRewards.Builder.recipe(ID))
+                                                    .withCriterion("has_dirt", hasItem(Blocks.DIRT))
+                                    )
                     )
-                    .withRewards(AdvancementRewards.Builder.recipe(ID))
-                    .withCriterion("has_dirt", hasItem(Blocks.DIRT))
-                )
-            )
-            .build(consumer, ID);
+                    .build(consumer, ID);
 
             ConditionalRecipe.builder()
                     .addCondition(
                             not(
-                                and(
-                                        not(modLoaded("minecraft")),
-                                        itemExists("minecraft", "dirt"),
-                                        FALSE()
-                                )
+                                    and(
```

Comment:
Again here and below you add some tabs, might be better/more correct but its still unecessary.